### PR TITLE
#21801 Removing redundant IdentifierAPI.delete call

### LINKED ISSUE #21801: Moving Folders
Folders should always have the knowable/consistant id.

If someone wants to "move" a folder, we should not update the folder record. Instead, we should create a new folder record that has the correct id generated and delete the old folder / identifier when done.

I believe we will also need to update the info in the permission/permission_reference table as well.

### PR DESCRIPTION
**Pull request by nollymar**
The line `APILocator.getIdentifierAPI().delete(id);` was removed when a folder is deleted because there is a trigger that internally deletes the reference on the `identifier` table. Only the IdentifierCache is flushed.

### CODE REVIEW

**Review by fabrizzio-dotCMS — APPROVED** [review]

File: dotCMS/src/main/webapp/html/portlet/ext/browser/view_browser_js_inc.jsp [hunk 1]
```diff
@@ -1328,17 +1328,15 @@ Structure defaultFileAssetStructure = CacheLocator.getContentTypeCache().getStru
...
-        setTimeout('reloadContent()',500);
     }
 
     function moveFolderCallback (response) {
+        reloadContent();
         if(response == '<%= UtilMethods.escapeSingleQuotes(LanguageUtil.get(pageContext, "Folder-moved")) %>'){
             BrowserAjax.getTree(null, initializeTree);
             showDotCMSSystemMessage(response);
         } else {
-            reloadContent ();
```
Comment (freddyucv) [inline dotCMS/src/main/webapp/html/portlet/ext/browser/view_browser_js_inc.jsp:1341]: why are you removing all this?

**Review by freddyucv — COMMENTED** [review]
any test?

File: dotCMS/src/main/java/com/dotmarketing/portlets/folders/business/FolderFactoryImpl.java [hunk 2]
```diff
@@ -89,7 +89,12 @@ protected void delete(final Folder folder) throws DotDataException {
 
            folderCache.removeFolder(folder, id);
 
-		   APILocator.getIdentifierAPI().delete(id);
+		   /*Folder reference is not explicitly deleted from the identifier table because there is
+		   a db trigger that executes the delete operation once the folder table is cleaned up.
+		   Only cache is flushed.
+		   Trigger name: folder_identifier_check_trigger
+		   */
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotmarketing/portlets/folders/business/FolderFactoryImpl.java:96]: is it  better put this in the javadoc?

File: dotCMS/src/main/webapp/html/portlet/ext/browser/view_browser_js_inc.jsp (hunk 1 above)
Comment (nollymar) [reply dotCMS/src/main/webapp/html/portlet/ext/browser/view_browser_js_inc.jsp:1341]: I didn't remove it. If you look above, I moved the call up

File: dotCMS/src/main/java/com/dotmarketing/portlets/folders/business/FolderFactoryImpl.java (hunk 2 above)
Comment (nollymar) [reply dotCMS/src/main/java/com/dotmarketing/portlets/folders/business/FolderFactoryImpl.java:96]: Done

**Comment by nollymar** [comment]
> any test?

There are already tests for the FolderAPI.delete. This change only fixes a race condition.

**Review by freddyucv — APPROVED** [review]

**Pull request merged**